src/nifgen/examples/nifgen_script.py

```diff
@@ -2,52 +2,26 @@
 
 import argparse
 import nifgen
-import numpy as np
-from scipy import signal
+
+import math
+import random
 import sys
 import time
 
-number_of_points = 256
-
-
-def calculate_sinewave():
-    time = np.linspace(start=0, stop=10, num=number_of_points)    # np.linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None)
-    amplitude = np.sin(time)
-    sinewave = amplitude.tolist()                               # List of Float
-    return sinewave
-
-
-def calculate_rampup():
-    ramp = np.linspace(start=0, stop=0.5, num=number_of_points)   # np.linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None)
-    ramp_up = ramp.tolist()                                     # List of Float
-    return ramp_up
-
-
-def calculate_rampdown():
-    ramp = np.linspace(start=0, stop=0.5, num=number_of_points)   # np.linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None)
-    ramp_down = ramp.tolist()                                   # List of Float
-    ramp_down.reverse()                                         # Reverse list to get a ramp down
-    return ramp_down
-
-
-def calculate_square():
-    time = np.linspace(start=0, stop=10, num=number_of_points)    # np.linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None)
-    square_build = signal.square(t=time, duty=0.5)              # signal.square(t, duty=0.5)
-    square = square_build.tolist()                              # List of Float
-    return square
-
-
-def calculate_triangle():
-    time = np.linspace(start=0, stop=1, num=number_of_points)     # np.linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None)
-    triangle_build = signal.sawtooth(t=time)                    # signal.sawtooth(t, width=1)
-    triangle = triangle_build.tolist()                          # List of Float
-    return triangle
+# waveform size should be a multiple of the quantum, which is 4, 2 or 1, for all devices
+# minimum waveform size varies with sample rate and device.
+# 512 should be enough for any device and sample rate, except certain 5450, 5451 configurations.
+NUMBER_OF_SAMPLES = 512
 
 
-def calculate_gaussian_noise():
-    random_noise = np.random.normal(loc=0, scale=0.1, size=number_of_points)  # random.normal(loc=0.0, scale=1.0, size=None)
-    noise = random_noise.tolist()                                           # List of Float
-    return noise
+# waveforms finish just short of 360 degrees, so that we don't repeat the first point
+# if we repeat the waveform
+SINE_WAVE = [math.sin(math.pi * 2 * x / (NUMBER_OF_SAMPLES)) for x in range(NUMBER_OF_SAMPLES)]
+RAMP_UP = [x / (NUMBER_OF_SAMPLES) for x in range(NUMBER_OF_SAMPLES)]
+RAMP_DOWN = [-1.0 * x for x in RAMP_UP]
+SQUARE_WAVE = [1.0 if x < (NUMBER_OF_SAMPLES / 2) else -1.0 for x in range(NUMBER_OF_SAMPLES)]
+SAWTOOTH_WAVE = RAMP_UP + [(-1 + x) for x in RAMP_UP]
```

Comment:
this constant does not have NUMBER_OF_SAMPLES samples.

```suggestion
SAWTOOTH_WAVE = RAMP_UP[::2] + [(-1 + x) for x in RAMP_UP][::2]
```

Reply:
Ah that's what you were doing with the slicing. Yeah, makes sense.